### ISSUE TITLE
Sign, verify fails for edwards448 and MDC201601

### ISSUE THREAD

**Issue by sundarnagarajan** [npmccallum/python-rubenesque#9]
I am not very knowledgeable about the mathematics of elliptic curve cryptography.

Works for curves except the following:
    edwards448
    edwards25519
    MDC201601

edwards25519 - I am not concerned, because there are numerous other implementations that support sign / verify

- Key generation works
- Generating shared session key works
- Signing 'seems to work', but verify fails
- Surprisingly, for MDC201601, verify seems to work 'sometimes' - I try generate, sign. verify cycle 10 times, and verify fails sometimes but not always for MDC201601

Sample code below:
This sample code also demonstrates [Issue 8](https://github.com/latchset/python-rubenesque/issues/8)
```
  import rubenesque.curves
  from rubenesque.signatures import ecdsa
  
  
  t = 'Hello world'
  
  not_working = ['edwards448', 'MDC201601', 'edwards25519']
  
  for curve_name in rubenesque.curves.supported():
      if curve_name in not_working:
          continue
      curve = rubenesque.curves.find(curve_name)
      pvt = curve.private_key() 
      pub = curve.generator() * pvt
      (r, s) = ecdsa.sign(curve.generator(), pvt, t)
      print('%s: Verified: %s' % (curve_name, ecdsa.verify(pub, t, r, s)))
  
  print('')
  
  curve_name = 'edwards448'
  curve = rubenesque.curves.find(curve_name)
  pvt = curve.private_key() 
  pub = curve.generator() * pvt
  (r, s) = ecdsa.sign(curve.generator(), pvt, t)
  print('%s: Verified: %s\n' % (curve_name, ecdsa.verify(pub, t, r, s)))
  
  print('MDC201601 cannot be found without explicit import')
  curve_name = 'MDC201601'
  try:
      curve = rubenesque.curves.find(curve_name)
  except Exception as e:
      print('Exception: ' + e.args[0])
      from rubenesque.curves.mdc import MDC201601       # noqa: F401
  
  curve = rubenesque.curves.find(curve_name)
  pvt = curve.private_key() 
  pub = curve.generator() * pvt
  (r, s) = ecdsa.sign(curve.generator(), pvt, t)
  print('%s: Verified: %s\n' % (curve_name, ecdsa.verify(pub, t, r, s)))
  
  n = 10
  print('Trying MDC201601 %d times\n' % (n))
  for x in range(n):
      curve_name = 'MDC201601'
      curve = rubenesque.curves.find(curve_name)
      pvt = curve.private_key() 
      pub = curve.generator() * pvt
      (r, s) = ecdsa.sign(curve.generator(), pvt, t)
      print('%d: %s: Verified: %s' % (
          x + 1, curve_name, ecdsa.verify(pub, t, r, s)))
```

**Comment by npmccallum** [comment]
First, I fixed #8.

Second, you don't want ECDSA signatures for Edwards curves (edwards25519, edwards448 and MDC201601). Instead, you want [EdDSA](https://en.wikipedia.org/wiki/EdDSA). We don't currently implement EdDSA. Pull requests to implement EdDSA are welcome.

**Comment by sundarnagarajan** [comment]
Yes, you're right - sorry for the elementary error - blame it on limited knowledge of mathematical cryptography
